### PR TITLE
added &stock command

### PR DESCRIPTION
**Pull request by Felix-44**
literally stock market with a new special currency
I may have forgot some stuff, thats why I would like a review

### CODE REVIEW

File: src/xilefunds.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            const price = args[1] || Stocks.value
+            if (price < 1) {
+                message.channel.send("uhh...you can't give this away for free...")
+                return
+            }
+            Stocks.auction.seller = message.author.username
+            Stocks.auction.sellerid = message.author.id
+            Stocks.auction.price = price
+            Stocks.timeout = setTimeout(() => {
+                const SellerDMChannel = message.author.createDM()
```
Comment (LJMSU) [inline src/xilefunds.js:94]: `User#createDM` returns a promise. You need to `.then` or `await` the value first.

File: src/xilefunds.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                if (Stocks.auction.buyer == undefined) {
+                    SellerDMChannel.send("It seems like nobody wanted to buy your xilefund, how sad...\n" +
+                        (Stocks.auction.price <= 50 ? "it was even at such a low price..." : "maybe you have to lower the price a little?"))
+                    return
+                }
+                const BuyerDMChannel = client.users.cache.get(Stocks.auction.buyerid).createDM()
```
Comment (LJMSU) [inline src/xilefunds.js:100]: `User#createDM` returns a promise. You need to `.then` or `await` the value first.

**Review by LJMSU — CHANGES_REQUESTED** [review]

File: src/xilefunds.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            let msg = "```diff\n"
+            const Ledger = Stocks.ledger
+            const ledgerlength = Ledger.length
+            let oldvalue = Ledger[ledgerlength - 11].price
+            for (let i = ledgerlength - 10; i < ledgerlength; i++) {
+                const Transaction = Ledger[i]
+                msg = `${msg}${Transaction.price > oldvalue ? "+" : "-"} ${Math.abs((Transaction.price - oldvalue) / oldvalue * 100).toFixed(2)}% => [${Transaction.price}] (${Transaction.seller} -> ${Transaction.buyer}) ${i == ledgerlength - 1 ? "CURRENT" : ""}\n`
+                oldvalue = Transaction.price
+            }
+            msg = msg + "```"
```
[the remark this answers is not in the document]
Comment (Felix-44) [reply src/xilefunds.js:39]: ah, I was unsure if that was possible so I used the usual way

File: src/xilefunds.js (hunk 1 above)
Comment (Felix-44) [reply src/xilefunds.js:94]: oh alright

**Comment by Felix-44** [comment]
@LJMSU I fixed all the issues, is there anything else left?

File: src/xilefunds.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+            Stocks.auction.price = price
+            Stocks.timeout = setTimeout(() => {
+                const SellerDMChannel = message.author.createDM().then(() => {
+                    if (Stocks.auction.buyer == undefined) {
+                        SellerDMChannel.send("It seems like nobody wanted to buy your xilefund, how sad...\n" +
+                            (Stocks.auction.price <= 50 ? "it was even at such a low price..." : "maybe you have to lower the price a little?"))
+                        return
+                    }
+                    const BuyerDMChannel = client.users.cache.get(Stocks.auction.buyerid).createDM().then(() => {
+                        const BuyerEconomySystem = Economy.getEconomySystem({ id: Stocks.auction.buyerid, username: Stocks.auction.buyer })
```
Comment (LJMSU) [inline src/xilefunds.js:101]: I think forgot to remove the `})`

**Review by LJMSU — CHANGES_REQUESTED** [review]

File: src/xilefunds.js (hunk 4 above)
Comment (Felix-44) [reply src/xilefunds.js:101]: uh..I dont see the issue

Comment (LJMSU) [reply src/xilefunds.js:101]: oh ok sorry

**Review by LJMSU — APPROVED** [review]

**Pull request merged**